### PR TITLE
corrscope: 0.7.0 -> 0.7.1

### PR DESCRIPTION
**Pull request by OPNA2608**
###### Motivation for this change
Bump. Includes some backwards-incompatible changes to command-line options so I prolly won't backport it to stable.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/corrscope/default.nix [hunk 1]
```diff
@@ -1,31 +1,28 @@
...
+    url = "https://github.com/corrscope/corrscope/releases/download/0.7.1/corrscope-${version}-py3-none-any.whl";
+    sha256 = "0xc6mp4b5msp4rgbzlvjypm3w1vw41r9h5hxvcw25lgj5h2yimhv";
   };
 
-  postPatch = ''
-    substituteInPlace setup.py \
-      --replace 'attrs>=18.2.0,<19.0.0' 'attrs>=18.2.0' \
-      --replace 'numpy>=1.15,<2.0,!=1.19.4' 'numpy>=1.15,<2.0'
-  '';
+  format = "wheel";
```
Comment (OPNA2608) [inline pkgs/applications/video/corrscope/default.nix:19]: https://github.com/corrscope/corrscope/commit/a3e779e13cf3d954737cc221755dd4222e6719d6#commitcomment-54263064

Comment (jonringer) [reply pkgs/applications/video/corrscope/default.nix:19]: source would be highly perferred, not sure how difficult it would be to just build from source

Comment (OPNA2608) [reply pkgs/applications/video/corrscope/default.nix:19]: It's built with poetry, I couldn't figure out how to package that so I initially went with pypi instead. I can check if something about out poetry support improved.

Comment (SuperSandro2000) [reply pkgs/applications/video/corrscope/default.nix:19]: I think we should build this from source, too. Shouldn't be that hard and probably just requires setting the format to poetry.

Comment (OPNA2608) [reply pkgs/applications/video/corrscope/default.nix]: `poetry2nix` was as confusing & not-working as before, but using `poetry-core` & pulling in an upstream patch works.

File: pkgs/applications/video/corrscope/default.nix [hunk 2]
```diff
@@ -1,31 +1,40 @@
...
+      sha256 = "0xxsbmxdbh3agfm6ww3rpa7ab0ysppan490w0gaqwmwzrxmmdljv";
+    })
+  ];
 
   nativeBuildInputs = [ wrapQtAppsHook ];
 
   buildInputs = [ ffmpeg qtbase ];
 
-  propagatedBuildInputs = with python3Packages; [ appdirs attrs click matplotlib numpy pyqt5 ruamel_yaml ];
+  propagatedBuildInputs = with python3Packages; [ appdirs atomicwrites attrs click matplotlib numpy poetry-core pyqt5 ruamel_yaml ];
```
Comment (SuperSandro2000) [inline pkgs/applications/video/corrscope/default.nix:37]: poetry-core should be in natibeBuildInputs.

**Pull request merged**